server/src/test/java/org/apache/druid/server/coordinator/duty/KillUnusedSegmentsTest.java

```diff
@@ -158,6 +159,14 @@ public void setup()
         overlordClient,
         config
     );
+    while (!target.canDutyRun()) {
```

Comment:
This doesn't seem clean. We shouldn't have to sleep in a loop in a test esp. when we are not actually verifying anything in or after the loop.

Instead, we should try to use config values in the test that ensure that the duty is ready to run right after it has been created.